### PR TITLE
[Merged by Bors] - chore: forward-port leanprover-community/mathlib#18962

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/GDelta.lean [hunk 1]
```diff
@@ -194,11 +192,36 @@ theorem isGδ_setOf_continuousAt [UniformSpace β] [IsCountablyGenerated (unifor
...
+theorem residual_of_dense_Gδ {s : Set α} (ho : IsGδ s) (hd : Dense s) : s ∈ residual α := by
+  rcases ho with ⟨T, To, Tct, rfl⟩
+  exact
+    (countable_sInter_mem Tct).mpr fun t tT =>
+      residual_of_dense_open (To t tT) (hd.mono (sInter_subset_of_mem tT))
+#align residual_of_dense_Gδ residual_of_dense_Gδ
+
+/-- A set is residual iff it includes a countable intersection of dense open sets. -/
+theorem mem_residual_iff {s : Set α} : s ∈ residual α ↔
+      ∃ S : Set (Set α), (∀ t ∈ S, IsOpen t) ∧ (∀ t ∈ S, Dense t) ∧ S.Countable ∧ ⋂₀ S ⊆ s :=
```
Comment (eric-wieser) [inline Mathlib/Topology/GDelta.lean:223]: This line wrapping is odd and doesn't match mathport
```suggestion
theorem mem_residual_iff {s : Set α} :
    s ∈ residual α ↔
      ∃ S : Set (Set α), (∀ t ∈ S, IsOpen t) ∧ (∀ t ∈ S, Dense t) ∧ S.Countable ∧ ⋂₀ S ⊆ s :=
```

**Review by eric-wieser — CHANGES_REQUESTED** [review]
I think you'll need to port https://leanprover-community.github.io/mathlib-port-status/file/topology/metric_space/baire in the same PR to make CI pass.

File: Mathlib/Topology/MetricSpace/Baire.lean
Comment (eric-wieser) [inline Mathlib/Topology/MetricSpace/Baire.lean:1]: This file is missing the SHA, import, and header comment changes.

Comment (eric-wieser) [reply Mathlib/Topology/MetricSpace/Baire.lean:1]: @ChrisHughes24, you resolved this but didn't push.

Comment (ChrisHughes24) [reply Mathlib/Topology/MetricSpace/Baire.lean:1]: I just pushed

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Pull request closed without merging**